app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1382,103 +1379,36 @@ public void onDialogDismissed(int dialogId) {
         // Nothing to do, no change made
     }
 
-    public static class DatePickerFragment extends DialogFragment
-            implements DatePickerDialog.OnDateSetListener {
-
-        public interface OnDatePickListener {
-            void onDatePicked(@NonNull LoyaltyCardField textFieldToEdit, @NonNull Date newDate);
-        }
-
-        private static final String TEXT_FIELD_TO_EDIT_ARGUMENT_KEY = "text_field_to_edit";
-        private static final String CURRENT_DATE_ARGUMENT_KEY = "current_date";
-        private static final String MIN_DATE_ARGUMENT_KEY = "min_date";
-        private static final String MAX_DATE_ARGUMENT_KEY = "max_date";
-        private static final String PICK_DATE_REQUEST_KEY = "pick_date_request";
-        private static final String NEWLY_PICKED_DATE_ARGUMENT_KEY = "newly_picked_date";
-
-        LoyaltyCardField textFieldEdit;
-        @Nullable
-        Date minDate;
-        @Nullable
-        Date maxDate;
-
-        public static DatePickerFragment newInstance(@NonNull LoyaltyCardField textField, @Nullable Date currentDate, @Nullable Date minDate, @Nullable Date maxDate) {
-            Bundle args = new Bundle();
-            args.putSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY, textField);
-            args.putSerializable(CURRENT_DATE_ARGUMENT_KEY, currentDate);
-            args.putSerializable(MIN_DATE_ARGUMENT_KEY, minDate);
-            args.putSerializable(MAX_DATE_ARGUMENT_KEY, maxDate);
-            DatePickerFragment fragment = new DatePickerFragment();
-            fragment.setArguments(args);
-            return fragment;
-        }
-
-        public static void registerDatePickListener(@NonNull AppCompatActivity activity, @NonNull OnDatePickListener listener) {
-            activity.getSupportFragmentManager().setFragmentResultListener(
-                    PICK_DATE_REQUEST_KEY,
-                    activity,
-                    (requestKey, result) -> listener.onDatePicked(
-                            (LoyaltyCardField) Objects.requireNonNull(result.getSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY)),
-                            (Date) Objects.requireNonNull(result.getSerializable(NEWLY_PICKED_DATE_ARGUMENT_KEY))));
-        }
-
-        @NonNull
-        @Override
-        public Dialog onCreateDialog(Bundle savedInstanceState) {
-            Bundle args = requireArguments();
-            textFieldEdit = (LoyaltyCardField) args.getSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY);
-            minDate = (Date) args.getSerializable(MIN_DATE_ARGUMENT_KEY);
-            maxDate = (Date) args.getSerializable(MAX_DATE_ARGUMENT_KEY);
-            // Use the current date as the default date in the picker
-            final Calendar c = Calendar.getInstance();
-
-            Date date = (Date) args.getSerializable(CURRENT_DATE_ARGUMENT_KEY);
-            if (date != null) {
-                c.setTime(date);
-            }
```

Comment:
This bit to use the existing value seems to have been lost. Maybe "current date" is a bit ambiguous though, since it refers to the currently *selected* date, not *right now*.

Reply:
Done